### ISSUE TITLE
fatal error: 'stdarg.h' file not found 

### ISSUE THREAD

**Issue by tiagomacarios** [include-what-you-use/include-what-you-use#369]
I am having the same symptom as #353, but it seems to be a different cause.

I have clang 3.9.1 installed from the nightly packages: http://apt.llvm.org/

I then installed iwyu from source:
```
git clone https://github.com/include-what-you-use/include-what-you-use.git
git checkout clang_3.9
mkdir build
cd build
cmake -DIWYU_LLVM_ROOT_PATH=/usr/lib/llvm-3.9 ..
make
make install
```

And now compile a simple example as in #353 give me the missing header error:
```
#include <iostream>

int main()
{
    std::cout << "Hi!\n";
}
```

```
$ include-what-you-use main.cpp
In file included from main.cpp:1:
In file included from /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/iostream:39:
In file included from /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/ostream:38:
In file included from /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/ios:38:
In file included from /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/iosfwd:40:
In file included from /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/bits/postypes.h:40:
In file included from /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/cwchar:44:
/usr/include/wchar.h:39:11: fatal error: 'stdarg.h' file not found
# include <stdarg.h>
          ^

main.cpp should add these lines:

main.cpp should remove these lines:
- #include <iostream>  // lines 1-1

The full include-list for main.cpp:

---
```

full log attached:
include-what-you-use -v -Xiwyu -v6 main.cpp
[log.txt](https://github.com/include-what-you-use/include-what-you-use/files/576510/log.txt)

**Comment by kimgr** [comment]
`stdarg.h` and the other compiler built-in headers come with Clang. This is a source of friction for IWYU, because we can't really install them on our own accord.

See https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/InstructionsForUsers.md#how-to-install and issue #100 for more details.

We've tried a few solutions with the built-in headers embedded into the IWYU binary, but it hasn't worked out for all target platforms. Ideas welcome!

**Comment by tiagomacarios** [comment]
Kim,

Thanks for the prompt response.

If I understood correctly for version 3.9(.1) I would need to have my
includes under: /usr/lib/clang/3.9/include correct?
If yes then it seems like my set-up is correct:

$ ls /usr/lib/clang/3.9/include | grep stdarg
stdarg.h

Any other ideas?

Tiago

**Comment by kimgr** [comment]
That sounds right. It depends on where IWYU is installed, but I suppose it's in /usr/bin?

What does `include-what-you-use -E -xc++ - -v < /dev/null` say?

**Comment by tiagomacarios** [comment]
Hum "make install" installed it to /usr/local/bin. Should I move it?

$ which include-what-you-use
/usr/local/bin/include-what-you-use

$ include-what-you-use -E -xc++ - -v < /dev/null
clang version 3.9.1-svn281634-1~exp1 (branches/release_39)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/local/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8.5
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9.4
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5.4.1
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.2.0
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.2.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
warning: argument unused during compilation: '-fsyntax-only'
clang invocation:
 "/usr/local/bin/include-what-you-use" "-cc1" "-triple"
"x86_64-pc-linux-gnu" "-E" "-disable-free" "-disable-llvm-verifier"
"-discard-value-names" "-main-file-name" "-" "-mrelocation-model" "static"
"-mthread-model" "posix" "-mdisable-fp-elim" "-fmath-errno" "-masm-verbose"
"-mconstructor-aliases" "-munwind-tables" "-fuse-init-array" "-target-cpu"
"x86-64" "-v" "-dwarf-column-info" "-debugger-tuning=gdb" "-resource-dir"
"/usr/local/bin/../lib/clang/3.9.1" "-internal-isystem"
"/usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0"
"-internal-isystem"
"/usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/x86_64-linux-gnu/c++/6.2.0"
"-internal-isystem"
"/usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/x86_64-linux-gnu/c++/6.2.0"
"-internal-isystem"
"/usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/backward"
"-internal-isystem" "/usr/local/include" "-internal-isystem"
"/usr/local/bin/../lib/clang/3.9.1/include" "-internal-externc-isystem"
"/usr/include/x86_64-linux-gnu" "-internal-externc-isystem" "/include"
"-internal-externc-isystem" "/usr/include" "-fdeprecated-macro"
"-fdebug-compilation-dir" "/mnt/c/_Working/passby/build/linux"
"-ferror-limit" "19" "-fmessage-length" "129" "-fobjc-runtime=gcc"
"-fcxx-exceptions" "-fexceptions" "-fdiagnostics-show-option" "-o" "-" "-x"
"c++" "-"

clang -cc1 version 3.9.1 based upon LLVM 3.9.1 default target
x86_64-pc-linux-gnu
ignoring nonexistent directory "/usr/local/bin/../lib/clang/3.9.1/include"
ignoring nonexistent directory "/include"
ignoring duplicate directory
"/usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/x86_64-linux-gnu/c++/6.2.0"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0
 /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/x86_64-linux-gnu/c++/6.2.0
 /usr/lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/backward
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
/mnt/c/_Working/t/include-what-you-use/iwyu_path_util.cc:92: Assertion
failed: !IsQuotedInclude(file_path)
Aborted (core dumped)

**Comment by kimgr** [comment]
Ah, at least that explains why it doesn't work -- the Clang libraries used by IWYU search for the built-in headers in dirname(executable)/../lib/clang/$version/include. Since IWYU is in /usr/local/bin and the Clang headers are in /usr/lib/..., they can't be found.

I'd probably copy the headers over manually to /usr/local/lib/clang/3.9.0/include (assuming that's the right version) if I were you.

I wish this was better...

**Comment by kimgr** [comment]
Sorry, make that `.../clang/3.9.1/...` since IWYU seems to be built against Clang 3.9.1.

**Comment by tiagomacarios** [comment]
Alright. I created a symbolic link and everything seems to work. Thanks!

Another question, is there any plans on have deb packages for 3.9? Seems
like I can only apt-get install 3.8-

**Comment by kimgr** [comment]
Good to hear it works better for you!

We don't do packaging ourselves, but we've had help in the past from various maintainers. I'm not sure who's done Debian before, I'm not a Linux user myself.

Now that the IWYU release for 3.9 is out, packaging should at least be possible.

**Comment by kimgr** [comment]
The immediate problem seems to be solved, closing as a duplicate of #100 and #353.
